eblocker-icapserver/src/main/java/org/eblocker/server/http/service/DeviceOnlineStatusCache.java

```diff
@@ -59,17 +59,38 @@ public void updateOnlineStatus(String deviceId) {
      *
      * @param device
      */
-    public void setOnlineStatus(Device device) {
-        if (device.isVpnClient()) {
-            device.setOnline(true);
-        } else {
-            Instant lastSeen = cache.get(device.getId());
-            if (lastSeen != null) {
-                Instant now = clock.instant();
-                device.setOnline(lastSeen.plusSeconds(deviceOfflineAfterSeconds).isAfter(now));
+    /**
+ * Sets the online status on the given device object based on the last seen timestamp from the cache.
+ *
+ * @param device the device to set the online status on
+ */
+public void setOnlineStatus(Device device) {
+    // VPN clients are always online
+    if (device.isVpnClient()) {
+        device.setOnline(true);
+    } else {
+        // Get the last seen timestamp from the cache
+        Instant lastSeen = cache.get(device.getId());
+        if (lastSeen != null) {
+            // Calculate the timestamp when this device should be considered offline
+            Instant offlineSince = lastSeen.plusSeconds(deviceOfflineAfterSeconds);
+            //if device is offline for more than 24 hours, calculate the days and put in in lastSeen
+            if (offlineSince.isBefore(clock.instant())) {
```

Comment:
When I switch off a device, the UI says (after 90 seconds) "Online: 0 days ago".

Reply:
fixed in eblocker-icapserver/src/main/java/org/eblocker/server/common/data/Device.java in 226816c462a1663efe2c0ef925f029f9e5369be7
adding new logic